### PR TITLE
Define test that checks that put sentence api returns a bad request status code if the content of the sentence does not match its given structure

### PR DESCRIPTION
**Pull request by jean553**

### CODE REVIEW

File: tests/test_put_sentence.rs [hunk 1]
```diff
@@ -227,3 +227,38 @@ fn test_put_sentence_structure_that_does_not_exist_returns_404() {
...
+
+    let client = reqwest::Client::new();
+
+    let url = format!(
+        "{}/sentences/{}/structure",
+        tests_commons::SERVICE_URL,
+        &sentence_uuid,
+    );
+    let response = client.put(&url)
+        .body("<sentence>Contains different words.</sentence>")
```
Comment (allan-simon) [inline tests/test_put_sentence.rs]: put some sub tags with attrbutes etc.  so that you cover most of the cases

File: src/one_sentence.rs [hunk 2]
```diff
@@ -143,9 +146,27 @@ fn edit_sentence_structure<'r>(
     }
 
     let row: String = rows.get(0).get("content");
-    let allowed_words = row.split(" ");
 
-    /* XXX: decode the XML here */
+    /* remove all spaces for easier comparison after parser iteration */
+    let allowed_structure = str::replace(&row, " ", "");
```
Comment (allan-simon) [inline src/one_sentence.rs]: Whitespace are significant

File: tests/test_put_sentence.rs [hunk 3]
```diff
@@ -182,7 +182,12 @@ fn test_put_sentence_structure_returns_200() {
         tests_commons::SERVICE_URL,
         sentence_uuid,
     );
-    let modified_structure = "<sentence>This is one sentence</sentence>";
+    let modified_structure = r#"
+        <subject>This</subject>
```
Comment (allan-simon) [inline tests/test_put_sentence.rs]: You're missing a root element so it should not be accepted + space are significant so you can't indent like in a programming language

File: tests/test_put_sentence.rs [hunk 4]
```diff
@@ -227,3 +232,38 @@ fn test_put_sentence_structure_that_does_not_exist_returns_404() {
...
 }
+
+#[test]
+fn test_put_sentence_structure_that_does_not_match_content_returns_400() {
+
+    let connection = db::get_connection();
+    db::clear(&connection);
+
+    let sentence_uuid = uuid::Uuid::new_v4();
+    let sentence_text = "This is one sentence";
```
Comment (allan-simon) [inline tests/test_put_sentence.rs]: Btw you're always missing the period in your sentences

File: tests/test_put_sentence.rs [hunk 5]
```diff
@@ -227,3 +232,38 @@ fn test_put_sentence_structure_that_does_not_exist_returns_404() {
...
+
+    let client = reqwest::Client::new();
+
+    let url = format!(
+        "{}/sentences/{}/structure",
+        tests_commons::SERVICE_URL,
+        &sentence_uuid,
+    );
+    let response = client.put(&url)
+        .body("<subject>I</subject><verb>eat</verb><complement>apple</complement>")
```
Comment (allan-simon) [inline tests/test_put_sentence.rs]: Spaces are missing

File: tests/test_put_sentence.rs (hunk 4 above)
Comment (jean553) [reply tests/test_put_sentence.rs]: fixed.

**Pull request merged**